src/components/VmActions/AutoSelectConsole.js

```diff
@@ -0,0 +1,111 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import { downloadVmConsoles } from '_/actions'
+import ConsoleConfirmationModal from '_/components/VmActions/ConsoleConfirmationModal'
+import {
+  EMPTY_CONSOLES_LIST,
+  NO_DEFAULT_CONSOLE,
+  DISPATCH_STATUSES,
+} from '_/constants'
+const {
+  DO_NOT_DISPATCH,
+  SHOULD_DISPATCH,
+  DISPATCHED,
+} = DISPATCH_STATUSES
+const defaultConsoleProtocolErrorsList = [
+  NO_DEFAULT_CONSOLE,
+  EMPTY_CONSOLES_LIST,
+]
+
+const AutoSelectConsole = ({
+  vm,
+  show,
+  isNoVNC,
+  fetchConsoles,
+  id,
+  onClose,
+}) => {
+  const initialState = {
+    selectedProtocol: undefined,
+    dispatchStatus: DO_NOT_DISPATCH,
+  }
+
+  const [state, setState] = useState(initialState)
+  const {
+    dispatchStatus,
+    selectedProtocol,
+  } = state
+
+  const handleClose = useCallback(() => {
+    console.log('handleClose')
+    setState(state => ({ ...state, dispatchStatus: DO_NOT_DISPATCH }))
+    onClose()
+  }, [])
+
+  useEffect(() => {
+    if (show) {
+      setState(state => ({
+        ...state,
+        dispatchStatus: SHOULD_DISPATCH,
+      }))
+      fetchConsoles(vm)
+    }
+  }, [show, fetchConsoles])
+
+  useEffect(() => {
+    if (show && dispatchStatus === SHOULD_DISPATCH) {
+      const selectedProtocol = vm.get('defaultConsole')
+
+      if (!selectedProtocol || selectedProtocol === 'undefined') {
+        return
+      }
+
+      if (defaultConsoleProtocolErrorsList.includes(selectedProtocol)) {
+        handleClose()
+      } else {
+        const consoles = vm.get('consoles')
+        setState({
+          selectedProtocol: consoles.find(console => console.get('protocol') === selectedProtocol),
```

Comment:
This seem to duplicate logic from `selectDefaultConsoleProtocol()`. Actually it seems to me that `vm.get('defaultConsole')` could be used directly as `props.selectedProtocol` without the need of storing it in local state (own props already contain `vm` so we can retrieve it in `connect()`).